### PR TITLE
Fix 407 type on no value elements

### LINKED ISSUE #407: Error when using type on an element without a value property
- `@testing-library/user-event` version: 12.0.11
- Testing Framework and version: jest - 26.1.0
  
- DOM Environment: react-dom - 16.13.1
  

Relevant code or config

```javascript
import React from "react";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

describe("test", () => {
  it("does not throw on div type", async () => {
    render(<div tabIndex={0}>foo</div>);
    try {
      screen.getByText("foo").focus();
      await userEvent.type(screen.getByText("foo"), " ");
    } catch (e) {
      expect(e).toEqual("");
    }
  });
});
```

Reproduction repository:

https://codesandbox.io/s/agitated-meitner-3zhxf?file=/src/user-event.test.js

Problem description:

I assume  that `userEvent.type` is the preferred way to simulate a keydown + keyup event on focused elements. If that is not the case, please consider this issue as irrelevant 😅 

Using type on a element without value should not throw an error.

Suggested solution:

- Don't throw an error when an element has no `value` property
- Handle value mutation only if an element has a property `value`

Reminder: Add suggestions from https://github.com/testing-library/user-event/pull/409

### PR DESCRIPTION
**Pull request by visualjerk**
**What**:

Closes #407 

**Why**:

Because typing on focusable elements should be possible without getting an error.

**How**:

Checked if an element has a value prop before handling input and selection on it.

**Checklist**:

- [x] Documentation (not needed)
- [x] Tests
- [x] Typings (not needed)
- [x] Ready to be merged

### CODE REVIEW

**Comment by marcosvega91** [comment]
I think that we are missing one thing. We should enable modifiers on `div` only if it is focusable ( only if has a tabindex)

**Comment by kentcdodds** [comment]
Good point. In fact, I think most events should not be fired on elements that aren't focusable. That said, I don't think the events *will* be fired on non-focusable elements anyway because we don't fire them on the given element, we fire them on the `element.ownerDocument.activeElement`, which could never be anything but a focusable element.

File: src/__tests__/type-modifiers.js [hunk 1]
```diff
@@ -497,15 +493,20 @@ test('{space} with preventDefault keydown on button', () => {
     button - mouseup: Left (0)
     button - click: Left (0)
     button - keydown: (32)
+    button - keypress: (32)
```
[the remark this answers is not in the document]
Comment (visualjerk) [reply src/__tests__/type-modifiers.js:526]: This change is intentional. If keydown prevents default, no keypress should be fired. However I missed that `click` should also not be fired in this case 😅

Good thing, you mentioned it!

[the remark this answers is not in the document]
Comment (visualjerk) [reply src/__tests__/type-modifiers.js:526]: Wait a moment ... with the way github displayed the comment, it looked like it was inside the `{space} with preventDefault keydown on button` test. What I wrote above was relating to that.

But inside the test you actually commented, the change is also expected. Missed the keypress in the previous PR.

**Pull request merged**